### PR TITLE
Add vertx v3.0.0/3.3.3 w/ git auto-update & update package.json

### LINKED ISSUE #9935: [Request] vertx-eventbus.js
Library name:  vertx-eventbus.js
npm package: https://registry.npmjs.org/vertx3-eventbus-client/-/vertx3-eventbus-client-3.3.3.tgz
Official homepage: http://vertx.io/docs/vertx-sockjs-service-proxy/java/
Super thanks, it has not been updated on the CND since version 2.0.

### PR DESCRIPTION
**Pull request by clairetsai818**
Since the GitHub repository of this library has changed, I've updated repository field and added auto-update field in package.json. Filename changed due to the authors of this library introduced a new breaking API that is not backward compatible so to signal this they use a new filename.

close #9935, cc @pmlopes

### CODE REVIEW

**Comment by clairetsai818** [comment]
@PeterDaveHello Please help me review this. Thanks.

**Comment by pmlopes** [comment]
@PeterDaveHello The filename change is due to the fact that with vert.x 3.3.3 we introduced a new breaking API that is not bakcward compatible so to signal this we use a new filename.

The repository url is just a bugfix since the type is "git" and the protocol was not git but http.

As the library developer I think these changes are good to me.

**Comment by PeterDaveHello** [comment]
Thanks @pmlopes ! Very appreciate your explanation, since @clairetsai818 is our intern program student, she is now learning how to send a good pull request, so I'll ask her to provide the more detail in the comment :smile:

**Comment by clairetsai818** [comment]
@PeterDaveHello I've revised the comment. Please help me review this, thanks!

File: ajax/libs/vertx/package.json [hunk 1]
```diff
@@ -17,6 +17,18 @@
   ],
   "repository": {
     "type": "git",
-    "url": "https://github.com/vert-x/vert.x"
+    "url": "git://github.com/vert-x3/vertx-bus-bower"
```
Comment (PeterDaveHello) [inline ajax/libs/vertx/package.json]: This seem to be a release purpose repo, that's not what we expected, you already use it at the auto-update config, this part should be a development purpose one.

Comment (pmlopes) [inline ajax/libs/vertx/package.json]: @clairetsai818 the development git repo lives here: `git://github.com/vert-x3/vertx-web` and the source file `vertx-web/vertx-web/src/client/vertx-eventbus.js`.

**Comment by clairetsai818** [comment]
@pmlopes Thanks for your help!
@PeterDaveHello I revise the file map and repo url, please help me review this, thanks!

**Comment by clairetsai818** [comment]
@PeterDaveHello I got all the old versions back. Please help me review this, thanks!

**Comment by PeterDaveHello** [comment]
@clairetsai818 You didn't answer my question.

**Comment by clairetsai818** [comment]
@PeterDaveHello I accidentally removed them, and I didn't notice that.

**Comment by PeterDaveHello** [comment]
@pmlopes just to make sure everything is good, I would like to confirm the repo details, as the origin/old repo [vert-x/vert.x](https://github.com/vert-x/vert.x)'s readme mentions, it's moved to [eclipse/vert.x](https://github.com/eclipse/vert.x), so what's the relationship between these two repos? Thanks.

**Comment by pmlopes** [comment]
@PeterDaveHello the vert.x project is a Eclipse Foundation project, however we have only moved the "core" java library to Eclipse, the process of moving the other components is still on going without an estimated ETA. So to summarize the repos:

* eclipse/vert.x - it is the java core vert.x
* vert-x3/* - are the vert.x v3.x repos that are the ecosystem of vert.x and where the vertx-eventbus.js lives
* vert-x/* -are the old repos vert.x v2.x (and these ones mark the time before the application to the Eclipse Foundation)

So all repo's pointing to "vert-x/*" should be considered legacy.

**Comment by PeterDaveHello** [comment]
@clairetsai818 why did you just changed the auto-update source repo? Shouldn't it be https://github.com/vert-x3/vertx-bus-bower? Maybe we could also consider to use npm package `vertx3-eventbus-client` as the auto-update source?

**Comment by pmlopes** [comment]
@PeterDaveHello @clairetsai818 both the npm package and the repo vert-x3/vertx-bus-bower copy the source js file from the vert-x3/vertx-web repo when we make a release. So to track the real source code when releases are tagged, looking to vert-x3/vertx-web should be the place. (But i'm not familiar with the auto-update features of cndjs)

**Comment by PeterDaveHello** [comment]
I prefer to use that npm or bower git repo, which has less overhead of auto-update process for us, and both of them are for release purpose, makes more sense.

**Comment by pmlopes** [comment]
@PeterDaveHello fine by vert.x team too. Since they are kept in sync with releases. Use the bower repo then since the npm comes from: https://github.com/vert-x3/vertx-stack but you'll never see the source there since it is an automated build that will at build time extract the file from the correct artifact and upload to npm.

**Comment by pmlopes** [comment]
@clairetsai818 if you update the `repository > url` to ` https://github.com/vert-x3/vertx-bus-bower` I think we're complete.

**Comment by PeterDaveHello** [comment]
@pmlopes The repo url shouldbbe the origin, but the autoupdate source should be updated 😉

File: ajax/libs/vertx/package.json [hunk 2]
```diff
@@ -17,6 +17,18 @@
   ],
   "repository": {
     "type": "git",
-    "url": "https://github.com/vert-x/vert.x"
+    "url": "git://github.com/vert-x3/vertx-web"
+  },
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/vert-x3/vertx-web.git",
```
Comment (pmlopes) [inline ajax/libs/vertx/package.json]: please update to: `git://github.com/vert-x3/vertx-bus-bower.git`

**Comment by pmlopes** [comment]
@PeterDaveHello thanks for the explanation. I've added a code review with the update

**Comment by PeterDaveHello** [comment]
Thank you @pmlopes

**Comment by clairetsai818** [comment]
@PeterDaveHello Please help me review this, thanks!

**Comment by clairetsai818** [comment]
@cdnjs/intern3 Please help me review this PR, thanks!

File: ajax/libs/vertx/package.json [hunk 3]
```diff
@@ -18,5 +18,17 @@
   "repository": {
     "type": "git",
     "url": "https://github.com/vert-x/vert.x"
```
Comment (sufuf3) [inline ajax/libs/vertx/package.json]: I think you should add another commit for update `package.json`.

Comment (PeterDaveHello) [inline ajax/libs/vertx/package.json]: no worry in this case 👍

Comment (extend1994) [inline ajax/libs/vertx/package.json]: @PeterDaveHello Can I ask how to determine what cases should we add another commit? (or should not)

Comment (PeterDaveHello) [inline ajax/libs/vertx/package.json]: depends on if it's part of the version update.

**Comment by clairetsai818** [comment]
@hare1039 I changed the source files, please help me review this again, thanks!

**Comment by maruilian11** [comment]
but you didn't rebase the branch with the latest master.

**Comment by PeterDaveHello** [comment]
@maruilian11 it's fine it didn't prevent you to review the other parts ... keep rebasing helps nothing, who knows if it will pass all the review or not?

**Comment by PeterDaveHello** [comment]
@clairetsai818 I thought you know about that you need to wrap the commit message, don't you?

**Comment by clairetsai818** [comment]
@maruilian11 I rebase this branch, please help me review this again, thanks!
@PeterDaveHello I wrapped the commit msg.

File: ajax/libs/vertx/package.json [hunk 4]
```diff
@@ -18,5 +18,17 @@
...
+  },
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/vert-x3/vertx-bus-bower.git",
+    "fileMap": [
+      {
+        "basePath": "",
+        "files": [
+          "vertx-eventbus.*"
```
Comment (maruilian11) [inline ajax/libs/vertx/package.json]: do you check `3.0.0` file structure?

Comment (maruilian11) [inline ajax/libs/vertx/package.json]: should be `vertx-eventbus*`, not `vertx-eventbus.*`.

**Comment by clairetsai818** [comment]
@maruilian11 I revised the filemap and manually added v3.0.0, please help me review this again, thanks!

**Comment by maruilian11** [comment]
@clairetsai818 please learn to use cdnjs autoupdate lib machanism to fetch files before sending the commits, then you will know why `vertx-eventbus*` can't fetch files we need in `3.0.0`. Thanks.

**Comment by maruilian11** [comment]
why cc the same issue twice in one commit, and don't use it in another one?

**Comment by PeterDaveHello** [comment]
Ping @hare1039 @sufuf3 @maruilian11

**Comment by maruilian11** [comment]
I think you can also update other infos in package.json by using another commit.
BTW, please use ADD not UPDATE in commit msg. you didn't update those versions, but you can use update after you update the infos.

**Comment by clairetsai818** [comment]
@maruilian11 Peter said I don't need to separate the commit in this PR, and I've revised the commit msg, please help me review this, thanks!

**Comment by clairetsai818** [comment]
@PeterDaveHello please help me review this, thanks!

**Comment by maruilian11** [comment]
@clairetsai818 I didn't ask you to separate the commit, other information is still old, like `repository`, `license` ...

**Comment by maruilian11** [comment]
`due to the the authors`

**Comment by PeterDaveHello** [comment]
Ping @clairetsai818

**Comment by pvnr0082t** [comment]
@clairetsai818 I think she just wants you to delete duplicated **`the`**.

**Comment by clairetsai818** [comment]
@pvnr0082t Thank you for pointing that out. @maruilian11 please help me review this again, thanks!

**Comment by PeterDaveHello** [comment]
@clairetsai818 next time you updated the PR, please select the reviewer again.

**Comment by clairetsai818** [comment]
@pvnr0082t I revised the commit msg, please help me review this, thanks!

**Comment by maruilian11** [comment]
the issue is asking for v3.3.3, don't use update info commit to close it.

**Comment by pmlopes** [comment]
Hi, just an heads-up, the vert.x project has been updated to 3.4.0 I've no idea if this affects your PR.

**Comment by PeterDaveHello** [comment]
@pmlopes Thanks

**Comment by clairetsai818** [comment]
@maruilian11 I revised the commit msg, please help me review this, thanks!

**Comment by clairetsai818** [comment]
@pvnr0082t I renamed the files in v3.0.0, please help me review this, thanks!

**Comment by PeterDaveHello** [comment]
@clairetsai818 Please ask review from your team mates first.

**Comment by maruilian11** [comment]
@PeterDaveHello Please take a look, thanks.

File: ajax/libs/vertx/package.json [hunk 5]
```diff
@@ -17,6 +14,18 @@
   ],
   "repository": {
     "type": "git",
-    "url": "https://github.com/vert-x/vert.x"
+    "url": "https://github.com/vert-x3/vertx-bus-bower"
```
Comment (PeterDaveHello) [inline ajax/libs/vertx/package.json]: https://github.com/cdnjs/cdnjs/pull/10416#pullrequestreview-18849713

File: ajax/libs/vertx/3.0.0/vertx-eventbus.js [hunk 6]
```diff
@@ -0,0 +1,228 @@
+/*
+ *   Copyright (c) 2011-2015 The original author or authors
+ *   ------------------------------------------------------
+ *   All rights reserved. This program and the accompanying materials
+ *   are made available under the terms of the Eclipse Public License v1.0
+ *   and Apache License v2.0 which accompanies this distribution.
+ *
+ *       The Eclipse Public License is available at
+ *       http://www.eclipse.org/legal/epl-v10.html
```
Comment (PeterDaveHello) [inline ajax/libs/vertx/3.0.0/vertx-eventbus.js]: Looks like `EPL-1.0 ` is still here? Can you please confirm that?

Comment (pmlopes) [inline ajax/libs/vertx/3.0.0/vertx-eventbus.js]: @PeterDaveHello all vert.x components are released as dual license EPL-1.0/ASL2.0

Comment (PeterDaveHello) [inline ajax/libs/vertx/3.0.0/vertx-eventbus.js]: I think @clairetsai818 removed it in package.json though I don't know why, seems to be added back now.

File: ajax/libs/vertx/package.json [hunk 7]
```diff
@@ -1,12 +1,12 @@
...
-  "version": "2.0.0",
+  "filename": "vertx-eventbus.min.js",
+  "version": "3.3.3",
   "description": "Effortless asynchronous application development for the modern web and enterprise. This library internally uses SockJS to send and receive data to a SockJS vert.x server called the SockJS bridge. http://vertx.io/core_manual_js.html#sockjs-eventbus-bridge",
   "homepage": "http://vertx.io/",
   "licenses": [
-    "EPL-1.0",
-    "Apache-2.0"
+    "Apache-2.0",
+    "EPL-1.0"
```
Comment (PeterDaveHello) [inline ajax/libs/vertx/package.json]: Any reason to change the license order here? I noticed that you removed and added EPL-1.0 back, and you should keep the minimum diff, the diff here is meaningless.

**Comment by clairetsai818** [comment]
@PeterDaveHello Please help me review this, thanks!

**Comment by maruilian11** [comment]
`Update license and repo info of vertx, cc #9935` ---> `Update repo info of vertx, cc #9935`

**Comment by clairetsai818** [comment]
@maruilian11 I revised the commit msg, please help me review this, thanks!

**Comment by maruilian11** [comment]
I think this PR is fine.
@cdnjs/intern2 Please help me double check this PR, thanks.

**Comment by PeterDaveHello** [comment]
The commit message should be wrapped ...

**Pull request merged**